SnakeGame/Snake.js

```diff
@@ -0,0 +1,207 @@
+class Snake {
+    constructor(game) {
+        this.game = game;
+        this.direction = "R";
+        this.squeue = new Queue();
+        this.head = [2, 3];
+        // start off as a 3 unit long snake
+        this.squeue.enqueue([2, 1]);
+        this.squeue.enqueue([2, 2]);
+        this.squeue.enqueue([2, 3]);
+        this.appleCount = 0;
+        this.input = "R";
+
+        pageCounter.textContent = "Apples Eaten: " + this.appleCount;
+    }
+
+    move() {
+        let dir = this.direction;
+        if (this.input == "R" && dir == "L"
+            || this.input == "L" && dir == "R"
+            || this.input == "U" && dir == "D"
+            || this.input == "D" && dir == "U") {
+        } else {
+            this.direction = this.input;
+        }
+        dir = this.direction;
+
+        if (dir == "R") {
+            this.squeue.enqueue([this.head[0], this.head[1] + 1]);
+            this.head = [this.head[0], this.head[1] + 1];
+        } else if (dir == "U") {
+            this.squeue.enqueue([this.head[0] - 1, this.head[1]]);
+            this.head = [this.head[0] - 1, this.head[1]];
+        } else if (dir == "L") {
+            this.squeue.enqueue([this.head[0], this.head[1] - 1]);
+            this.head = [this.head[0], this.head[1] - 1];
+        } else if (dir == "D") {
+            this.squeue.enqueue([this.head[0] + 1, this.head[1]]);
+            this.head = [this.head[0] + 1, this.head[1]];
+        }
+        let tail;
+
+        // check if apple is eaten
+        let apple = game.apple;
+        if (this.head[0] == apple[0] && this.head[1] == apple[1]) {
+            tiles[apple[0]][apple[1]].classList.remove("apple");
+            this.appleCount++;
+            this.game.apple = game.newAppleLoc();
+            pageCounter.textContent = "Apples Eaten: " + this.appleCount;
+        } else {
+            tail = this.squeue.dequeue().data;
+        }
+
+        game.tick(this.head, tail);
+    }
+}
+
+class Game {
+    constructor(n) {
+        this.n = n;
+    }
+
+    reset() {
+        this.board = [];
+        for (let i = 0; i < this.n; i++) {
+            let row = [];
+            for (let j = 0; j < this.n; j++) {
+                row.push('-');
+            }
+            this.board.push(row);
+        }
+
+        this.snake = new Snake(this);
+        this.gameState = "NOT_STARTED";
+
+        // reset tiles on page
+        let allTiles = document.getElementsByClassName("snakeBody");
+        while (allTiles.length > 0) {
```

Comment:
You can use `.forEach` rather than a `while` loop, makes it a bit cleaner. Honestly, I'm surprised that this works (I'm surprised that `allTiles.length` will be updated if you remove elements from the class).